src/sshagent/ASN1Key.h

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright (C) 2018 Toni Spets <toni.spets@iki.fi>
+ *  Copyright (C) 2018 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef ASN1KEY_H
+#define ASN1KEY_H
+
+#include "OpenSSHKey.h"
+#include <QtCore>
+
+class ASN1Key
+{
+public:
+    static bool parseDSA(QByteArray& ba, OpenSSHKey& key);
+    static bool parseRSA(QByteArray& ba, OpenSSHKey& key);
+
+private:
+    static const quint8 TAG_INT        = 0x02;
+    static const quint8 TAG_SEQUENCE   = 0x30;
+    static const quint8 KEY_ZERO       = 0x0;
+
+    ASN1Key() { }
```

Comment:
= default or remove entirely as the default constructor is implied.

Reply:
Wouldn't an implied constructor be public? Went with `= default` for now.